src/aleph/web/controllers/programs.py

```diff
@@ -0,0 +1,39 @@
+import json
+from typing import Any, Dict, List
+
+from aiohttp import web
+from aleph_message.models import MessageType
+from bson import json_util
+
+from aleph.model.messages import Message
+from aleph.web import app
+
+
+async def get_programs_on_message(request: web.Request):
```

Comment:
Could you validate the request with Pydantic?

Reply:
Done, but I am not fan of the added complexity to just parse a number.